toptal/picasso
# fix: hide ShowMore if not needed

[FX-3351]

### Description

Do not display "Show more" button if there's nothing to show.

### How to test

- check [temploy example](https://picasso.toptal.net/fix-3351-dont-display-showmore-when-nothing-to-show/?path=/story/components-showmore--showmore#with-short-text) and try to play with content length and rows

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| <img width="1190" alt="image" src="https://user-images.githubusercontent.com/22159416/208976248-18a24d16-9f85-46db-abc2-efd6b7afc2dc.png"> | <img width="1184" alt="image" src="https://user-images.githubusercontent.com/22159416/208976291-e3c8e65c-4938-49f5-97b8-53d8bf8ce575.png"> |

### Development checks

- Annotate all `props` in component with documentation

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-3351]: https://toptal-core.atlassian.net/browse/FX-3351?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
